### PR TITLE
ci: validate async delete

### PR DESCRIPTION
**Pull request by camrynl**
**Reason for Change**:

- adds a script to validate pod async delete when cns is down
- only testing in swift for now, will add to cilium stages once azure-ipam changes are reverted back

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by rbtr — APPROVED** [review]
ty for adding this 👏 lgtm

**Comment by rbtr** [comment]
cc @tyler-lloyd

File: hack/scripts/async-delete-test.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    busybox_pod=$(kubectl get pods -l k8s-app=busybox -o wide | grep "$node_name" | awk '{print $1}')
+    if [ -z $busybox_pod  ]; then
+        continue
+    else
+        echo "wait 1 min for delete to processes and error to catch. expect a file to be written to var/run/azure-vnet/deleteIDs"
+        kubectl delete pod $busybox_pod 
+        sleep 60s
+        
+        echo "restore azure-cns pods"
+        kubectl -n kube-system patch daemonset azure-cns --type json -p='[{"op": "remove", "path": "/spec/template/spec/nodeSelector/non-existing"}]'
```
Comment (jpayne3506) [inline hack/scripts/async-delete-test.sh:21]: If the test fails to delete the pod does CNS get left in an unready state? Would like to be able to run the test without having to manually remove the patch.

Comment (camrynl) [reply hack/scripts/async-delete-test.sh:21]: updated the e2e step with a check to remove the patch if it still exists

**Review by jpayne3506 — APPROVED** [review]

**Review by jpayne3506 — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**